src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/NativeAllocationScope.java

```diff
@@ -226,6 +240,25 @@ public MemoryAddress allocate(long bytesSize) {
      */
     public abstract MemoryAddress allocate(long bytesSize, long bytesAlignment);
 
+    /**
+     * Claim ownership of a given segment. The input segment must be closeable - that is, it must feature
+     * the {@link MemorySegment#CLOSE} access mode. As a side-effect, the input segment will be marked
+     * as <em>not alive</em>, and a new segment will be returned.
+     * <p>
+     * The returned segment will feature only {@link MemorySegment#READ} and
+     * {@link MemorySegment#WRITE} access modes (assuming these were available in the original segment). As such
+     * the resulting segment cannot be closed directly using {@link MemorySegment#close()} - but it will be closed
+     * indirectly when this allocation scope is closed.
+     * @param segment the segment whose ownership will be claimed by this allocation scope.
+     * @return a new, non closeable memory segment, backed by the same underlying region as {@code segment},
+     * but whose life-cycle is tied to that of this allocation scope.
+     * @throws IllegalStateException if {@code segment} is not <em>alive</em> (see {@link MemorySegment#isAlive()}).
+     * @throws NullPointerException if {@code segment == null}
+     * @throws IllegalArgumentException if {@code segment.ownerThread() != this.ownerThread()}, or if {@code segment}
+     * does not feature the {@link MemorySegment#CLOSE} access mode.
+     */
+    public abstract MemorySegment claim(MemorySegment segment);
+
```

Comment:
I am not sure "claim" conveys the intent. Frankly I like "register" better.  In a way that name is consistent with ref   "Cleaner" API - user 'register's the reference for cleanup later.